### ISSUE TITLE
LSTM-CRF is worse than LSTM

### ISSUE THREAD

**Issue by kaituoxu** [glample/tagger#4]
hi, @glample 
Recently, I have modified your code in `model.py` and `train.py` to adapt to my sequence label mission, punctuation prediction, without modifying your kernel code in `nn.py`.

The same setup of LSTM and LSTM-CRF is word_dim=100, word_lstm_dim=100,  dropout=0.5, lr_method=adadelta,  char_dim = 0, without using char embedding.
Then I got unexpected results in which LSTM-CRF is worse than LSTM.

I'm wondering whether it is the influence of dropout layer or not? Cause I don't use char embedding, do I need to use dropout layer?

Or maybe LSTM-CRF is not suitable for my specified mission?

Do you have any suggestions?

**Comment by Justin1904** [comment]
Did you encounter any errors when setting char_dim=0? when I do so an error will come up in the first place, as described in #18 . I wonder how your settings will work in the first place.

**Comment by xuewei4d** [comment]
It seems there is a little difference in CRF [likelihood](https://github.com/glample/tagger/blob/master/nn.py#L258) loss from in [TensorFlow](https://github.com/tensorflow/tensorflow/blob/6b1d4fd8090d44d20fdadabf06f1a9b178c3d80c/tensorflow/contrib/crf/python/ops/crf.py#L272) . The unary potentials are in log_sum_exp, while in TF, it is outside.
